src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -693,54 +682,78 @@ class BinaryCompactObject : public DomainCreator<3> {
 
   Domain<3> create_domain() const noexcept override;
 
-  std::vector<std::array<size_t, 3>> initial_extents() const noexcept override;
+  std::vector<std::array<size_t, 3>> initial_extents() const noexcept override{
+    return initial_number_of_grid_points_;
+  }
+
+  std::vector<std::array<size_t, 3>> initial_refinement_levels()
+      const noexcept override {
+    return initial_refinement_;
+  }
 
-  std::vector<std::array<size_t, 3>> initial_refinement_levels() const
-      noexcept override;
+  std::vector<std::string> block_names() const noexcept override {
+    return block_names_;
+  }
+
+  std::unordered_map<std::string, std::unordered_set<std::string>>
+  block_groups() const noexcept override {
+    return block_groups_;
+  }
 
   auto functions_of_time() const noexcept -> std::unordered_map<
       std::string,
       std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>> override;
 
  private:
-  void check_for_parse_errors(const Options::Context& context) const;
-  void initialize_calculated_member_variables() noexcept;
-
   Object object_A_{};
   Object object_B_{};
   double radius_enveloping_cube_{};
   double radius_enveloping_sphere_{};
-  size_t initial_refinement_{};
-  size_t initial_grid_points_per_dim_{};
+  std::vector<std::array<size_t, 3>> initial_refinement_{};
+  std::vector<std::array<size_t, 3>> initial_number_of_grid_points_{};
   static constexpr bool use_equiangular_map_ =
       false;  // Doesn't work properly yet
   bool use_projective_map_ = true;
   bool use_logarithmic_map_outer_spherical_shell_ = false;
-  size_t addition_to_outer_layer_radial_refinement_level_{};
   double projective_scale_factor_{};
   double translation_{};
   double length_inner_cube_{};
   double length_outer_cube_{};
   size_t number_of_blocks_{};
   std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>
       outer_boundary_condition_;
+  std::vector<std::string> block_names_{};
+  std::unordered_map<std::string, std::unordered_set<std::string>>
+      block_groups_{};
 
   // Variables for FunctionsOfTime options
-  bool enable_time_dependence_;
-  double initial_time_;
-  std::optional<double> initial_expiration_delta_t_;
-  double expansion_map_outer_boundary_;
-  double initial_expansion_;
-  double initial_expansion_velocity_;
+  bool enable_time_dependence_{false};
```

Comment:
Did the member variables actually change? Having had the defaults in their own commit would make this easy to review